nerfstudio/models/nerfacto.py

```diff
@@ -163,6 +166,9 @@ def populate_modules(self):
             implementation=self.config.implementation,
         )
 
+        self.camera_optimizer: CameraOptimizer = self.config.camera_optimizer.setup(
+            num_cameras=self.num_train_data, device="cpu"  # for now initialize on cpu, these will be moved later
```

Comment:
This is very confusing for me. The camera_optimizer.device attribute is set at initialization as "cpu". But during training, attributes like self.pose_adjustment are moved to the gpu, so internal attributes are no longer on the "self.device" device. Is this intended? Should we somehow check that self.device changes inside the camera optimizer when e.g. self.pose_adjustment tensors get moved to the gpu?

Reply:
> Yes this device behavior has always been confusing to me too, I'm not sure what happens to the device parameter when you call .cuda() on the an nn.Module, does it also get updated to the gpu?

All parameters within a module will go to "gpu" but this user defined "self.device" param will stay as "cpu". It is not so important, just noticed that I could not rely on the self.device var when considering where I should send (gpu/cpu) my "non_trainable_indices" ints. I will check out your parallel data manager, awesome work!